app/models/message.rb

```diff
@@ -2,13 +2,20 @@ class Message < ActiveRecord::Base
   belongs_to :user
 
   validates_presence_of :to_phone_number, :body
+  validate :user_quota_limit
+
 
   after_commit :send_sms_job
 
   attr_reader :group_id
 
+  def user_quota_limit
+    self.user.quota_limit_check
+    errors.add(:message, "would exceed your quota. So it cannot be sent.")
```

Comment:
That's really great feedback. 

Thanks for that. I'll throw that on my list to do. Being self-taught and working on my own hasn't provided a lot of chances for me to have my code reviewed so I'm happy when I get any sort of feedback. 

By all means, feel free to review my code more :) Always looking for improvement!